Dockerfile

```diff
@@ -38,7 +38,6 @@ RUN addgroup vault && \
 
 # Set up certificates, base tools, and software.
 RUN set -eux && \
-    apk update && apk upgrade libretls && \
     apk add --no-cache ca-certificates libcap su-exec iputils
```

Comment:
Based on your PR description, I guess it still works, but should we keep the `apk update` command to ensure we get the latest for these packages?

Reply:
I feel like that could make the build slightly less reproducible? Though since this Dockerfile is referencing "alpine:latest" instead of a specific version, reproducible is probably a moot point here, so I'll keep the update call.

(For context, the whole line was introduced just to fix the libretls issue here: https://github.com/hashicorp/vault-k8s/pull/327/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R41)